src/Persistence/PersistentObjectFactory.php

```diff
@@ -290,30 +291,40 @@ protected function normalizeParameter(string $field, mixed $value): mixed
         if ($value instanceof self) {
             $pm = Configuration::instance()->persistence();
 
-            $inversedRelationshipMetadata = $pm->inverseRelationshipMetadata(static::class(), $value::class(), $field);
+            $relationshipMetadata = $pm->inverseRelationshipMetadata(static::class(), $value::class(), $field);
 
-            // handle inversed OneToOne
-            if ($inversedRelationshipMetadata && !$inversedRelationshipMetadata->isCollection) {
-                $inverseField = $inversedRelationshipMetadata->inverseField;
+            // handle inverse OneToOne
+            if ($relationshipMetadata && !$relationshipMetadata->isCollection) {
+                $inverseField = $relationshipMetadata->inverseField;
 
-                $inversedObject = $value->withPersistMode(
+                $value = $value->withPersistMode(
                     $this->isPersisting() ? PersistMode::NO_PERSIST_BUT_SCHEDULE_FOR_INSERT : PersistMode::WITHOUT_PERSISTING
-                )
+                );
+
+                if ((new \ReflectionClass(static::class()))->getProperty($field)->getType()?->allowsNull()) {
```

Comment:
I'm wondering if checking the "nullability" based on type system only is OK: it would not work with relationships which are typed with phpdoc only...

I think it's OK like this, I don't see any valid case where a one-to-one should not be strongly typed :thinking: 

Maybe we could use `symfony/typeinfo` for this?

I'm also wondering if we should reflect this in the exception message? "if the properties of the relations are not strongly types, please do..."